src/main/java/uk/ac/cam/cl/dtg/isaac/api/IsaacController.java

```diff
@@ -189,6 +189,12 @@ public final Response search(@Context final Request request, @Context final Http
             @DefaultValue(DEFAULT_START_INDEX_AS_STRING) @QueryParam("start_index") final Integer startIndex,
             @DefaultValue(DEFAULT_SEARCH_RESULT_LIMIT_AS_STRING) @QueryParam("limit") final Integer limit) {
 
+        // Impose 1000 character search string limit internally
+        if (searchString.length() > SEARCH_TEXT_CHAR_LIMIT) {
```

Comment:
Does IntelliJ think this is safe? Is it that it wouldn't have matched the endpoint if searchString was null, and so you don't need a null check?

Reply:
It infers the @notnull annotation - searchString has to be a string or it throws a 'can't find resource' when you try and hit it